crates/cargo-codspeed/src/build.rs

```diff
@@ -114,6 +171,22 @@ impl BuildOptions<'_> {
             exit(status.code().expect("Could not get exit code"));
         }
 
+        if !bench_targets_with_default_harness.is_empty() {
+            let targets_list = bench_targets_with_default_harness
+                .into_iter()
+                .map(|(package, bench)| format!("  - `{bench}` in package `{package}`"))
+                .collect::<Vec<_>>()
+                .join("\n");
+
+            bail!(
+                "The following benchmark targets use the default test harness:\n{targets_list}\n\n\
+                 Codspeed requires benchmark targets to disable the default test harness.\n\
+                 Either disable the default harness by adding `harness = false` to the corresponding\n\
+                 `[[bench]]` section in the Cargo.toml, or filter these benchmark targets.\n\
+                 See `cargo codspeed build --help` for more information.",
+            );
```

Comment:
The impact should be the first thing mentioned before the how or why. As a user you are more interested in this first.
Start by stating that codspeed will not work for the following targets, then explain why really quickly and finally give the two ways to fix it.
It can be a bit more structured and directly explain how to exclude packages.

Start with something, then let me know and we can iterate quickly on this